### PR TITLE
[BugFix]fix fe restart failed when executing drop external table on sr before dropping table on hive

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #
https://github.com/StarRocks/starrocks/issues/13770

you can reproduce this case by steps:
1. drop hive external table in sr.
2. drop hive table in hive.
3. restart fe
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 11 (18.18%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/CacheUpdateProcessor.java|2|11|18.18%|[166, 167, 168, 178, 179, 180, 181, 182, 183]|

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/CacheUpdateProcessor.java [hunk 1]
```diff
@@ -160,14 +160,28 @@ public void invalidateAll() {
     }
 
     public void invalidateTable(String dbName, String tableName) {
-        String tableLocation = ((HiveMetaStoreTable) metastore.getTable(dbName, tableName)).getTableLocation();
+        String tableLocation;
+        try {
+            tableLocation = ((HiveMetaStoreTable) metastore.getTable(dbName, tableName)).getTableLocation();
+        } catch (Exception e) {
+            LOG.warn("Failed to get table {}.{}. ignore it", dbName, tableName);
+            return;
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/CacheUpdateProcessor.java:168]: Just return, the table cache or partition cache may still exists?

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/hive/CacheUpdateProcessor.java:168]: yes, this happens during the restart. the cache is empty.  if delete the hive table first and then delete the hive external table. we will get table from cache. if the cache miss, others cache entry of this table are also empty.

**Review by gengjun-git — DISMISSED** [review]

**Review by Youngwb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 9 / 27 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|0|4|00.00%|[166, 167, 168, 169]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|0|1|00.00%|[347]|
|:large_blue_circle:|com/starrocks/connector/hive/events/DropTableEvent.java|0|1|00.00%|[72]|
|:large_blue_circle:|com/starrocks/connector/hive/CacheUpdateProcessor.java|4|16|25.00%|[186, 187, 188, 189, 191, 192, 206, 207, 208, 209, 210, 211]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|1|100.00%|[]|

**Review by Smith-Cruise — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Comment by dulong41** [comment]
run starrocks_be_unittest

**Comment by dulong41** [comment]
run starrocks_clang-tidy

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**